lib/node_modules/@stdlib/array/base/cuany-by/lib/assign.js

```diff
@@ -0,0 +1,322 @@
+			flg = true;
+		}
+		yset( ydata, io, flg );
+		io += stride;
+	}
+	return ydata;
+}
+
+/**
+* Cumulatively tests whether at least one element in a provided array passes a test implemented by a predicate function.
+*
+* @private
+* @param {Collection} x - input array
+* @param {Object} y - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @returns {Collection} output array
+*
+* @example
+* function isPositive( v ) {
+*     return v > 0;
+* }
+*
+* var Uint8Array = require( '@stdlib/array/uint8' );
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Uint8Array( [ false, false, false, true, false ] );
+* var y = toAccessorArray( [ false, null, false, null, false, null, false, null, false, null ] );
+*
+* var arr = boolean( x, arraylike2object( y ), 2, 0, isPositive );
+*
+* var v = y.get( 0 );
+* // returns false
+*
+* v = y.get( 2 );
+* // returns false
+*
+* v = y.get( 4 );
+* // returns false
+*
+* v = y.get( 6 );
+* // returns true
+*
+* v = y.get( 8 );
+* // returns true
+*/
+function boolean( x, y, stride, offset, predicate, thisArg ) {
+	var ydata;
+	var yset;
+	var flg;
+	var io;
+	var i;
+
+	yset = y.accessors[ 1 ];
+	ydata = y.data;
+
+	flg = false;
+	io = offset;
+	for ( i = 0; i < x.length; i++ ) {
+		if ( flg === false && predicate.call( thisArg, x[ i ], i, x ) ) {
+			flg = true;
+		}
+		yset( ydata, io, flg );
+		io += stride;
+	}
+	return ydata;
+}
+
+
+// MAIN //
+
+/**
+* Cumulatively tests whether at least one element in a provided array passes a test implemented by a predicate function and assigns results to provided output array.
+*
+* @param {Collection} x - input array
+* @param {Collection} y - output array
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @param {Function} predicate - test function
+* @param {*} [thisArg] - execution context
+* @returns {Collection} output array
+*
+* @example
+* function isPositive( v ) {
+*     return v > 0;
+* }
+*
+* var x = [ 0, 0, 0, 1, 0 ];
+* var y = [ false, null, false, null, false, null, false, null, false, null ];
+*
+* var out = assign( x, y, 2, 0, isPositive );
+// returns [ false, null, false, null, false, null, true, null, true, null ]
+*
+* var bool = ( y === out );
+* // returns true
+*/
+function assign( x, y, stride, offset, predicate, thisArg ) {
+	var xo = arraylike2object( x );
+	var yo = arraylike2object( y );
+
+	if (
+		xo.accessorProtocol ||
+		yo.accessorProtocol
+	) {
+		// If provided a complex number array, reinterpret as a real typed array and test interleaved real and imaginary components, where we consider a complex number to be truthy if at least one component is non-zero...
```

Comment:
We will need to update this comment, as it's not applicable here where we have a `predicate` function.